sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,661 @@
+
+IMPORTANT: In the current AMX and DPAS implementation, the layout in the load of matrix B (provided by the `layout memL` parameter below) must be `packed` or `row_major`. Automatic VNNI transform is supported on AMX. The layout in the load of matrices A and C must be `row_major`, and the layout in the store of matrix C (provided by the `layout memL` parameter below) must also be `row_major`.
+
+Since the matrix functions are group operations (as defined in Section 4.17.3 of the SYCL specification), the matrix API has to be accessed by all the work-items in the group in a convergent control flow. The `Group` template argument can be a work-group or a subgroup. These functions will be called once by each work item in the group.
+
+To be aligned with the SYCL 2020 group algorithms, an additional group argument is added to the matrix operations to designate that these functions are collective operations. The {dpcpp} syntax is the following: 
+
+IMPORTANT: In the current implementation, only the subgroup scope is supported.  
+
+#### Load
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          use Use, access::address_space Space>
+  void joint_matrix_load(Group sg,
+    joint_matrix<T, Use, NumRows, NumCols, layout::dynamic, Group> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride, layout memL);
+    
+  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          use Use, layout Layout, access::address_space Space>
+  void joint_matrix_load(Group sg,
+    joint_matrix<T, Use, NumRows, NumCols, Layout, Group> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride);
+}
+```
+
+`joint_matrix_load` loads data from memory to the 2d tiles/registers of Intel AMX/DPAS.
+We define two overloads of the load function depending on whether the memory layout was declared as part of the `joint_matrix` type or not. 
+The first overload that takes memory layout as an argument is only available for a `joint_matrix` type that was declared with `layout::dynamic`.
+The second overload without a memory layout must not be used with a `joint_matrix` type that was declared with `layout::dynamic`.
+
+#### Store
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          use Use, layout Layout, access::address_space Space>
+  void joint_matrix_store(Group sg,
+    joint_matrix<T, Use, NumRows, NumCols, Layout, Group> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride, layout memL);
+}
```

Comment:
No need for two versions. In both Intel and CUDA backends, store must take layout as a dynamic argument.
@JackAKirk please confirm

Reply:
What if the template parameter of the matrix is something other than `dynamic`, though?  For example, consider the case when `Layout` is `row_major`.  Does it still make sense to require the user to specify a layout parameter at runtime to `joint_matrix_store`?  What if the layout they specify at runtime is different from the `Layout` template parameter?  Is that OK?

Reply:
In the CUDA case, you can only store to matrix accumulator.
For the Intel backend, Besides C, you can store to matrix A or B. In this case, if users decide to do that, they assume it is not portable code, so they would use the version of joint matrix where A and B type layouts are dynamic.

Reply:
This is not something we needed to consider in the Nvidia backend because the nvidia backend doesn't support native implementations of `joint_matrix_store` for anything other than accumulator (and we don't plan on implementing it at all). My understanding is that the intel backend may want to support `joint_matrix_store` for use::a and use::b and that it can choose whatever interface it wants for this overload of `joint_matrix_store`. I thought that this would eventually be added to an intel specific extension document, similar to what we did for cuda in https://github.com/intel/llvm/pull/6968.